projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis-crosshair.ts

```diff
@@ -28,7 +28,8 @@ export class CartesianAxisCrosshair {
     this.updatePosition(
       crosshairSelection,
       position,
-      locationData.map(data => data.location)
+      locationData.map(data => data.location),
+      Array.isArray(locationData?.[0].dataPoint)
```

Comment:
We should isolate this logic in a separate method. No need to repeat it everywhere. 

Imo, this should be controlled via an input property and set from outside the generic cartesian chart. But

Reply:
Problem is that I'm checking the data here. I should follow a better logic